syntax = "proto3";

//import "login_id.proto";

package LoginProtocol;

message GameInfo {
    int32 gameid = 1;
    int32 platform = 2;
    string name = 3;
    string appid = 4;
    string secret = 5;
}

message C2S_LoginLogin {
    string loginname = 1;
}

message S2C_LoginLogin {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_LoginCreate {
    int32 gameid = 1;
    string loginname = 2;
    string role_name = 3;
    string avatarurl = 4;
}

message S2C_LoginCreate {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_LoginLoad {
    int64 role_id = 1;
    int32 gameid = 2;
    string loginname = 3;
}

message S2C_LoginLoad {
    int32 ret = 1;
    int64 role_id = 2;
    int32 gameid = 3;
    string loginname = 4;
}

message C2S_LoginOpenid {
    int32 gameid = 1;
    string code = 2;
}

message S2C_LoginOpenid {
    int32 ret = 1;
    string openid = 2;
}

message C2S_LoginUpdate {
    string role_name = 1;
    string avatarurl = 2;
}

message S2C_LoginUpdate {
    int32 ret = 1;
}